tasks/push-snapshot-to-quay/push-snapshot-to-quay.yaml

```diff
@@ -0,0 +1,109 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: push-snapshot-to-quay
+spec:
+  description: >-
+    Tekton task to push snapshot images to quay.io using `cosign copy`
+  params:
+    - name: releasePlan
+      description: Namespaced name of release plan - should be in format "namespace/name"
+      type: string
+    - name: snapshot
+      description: Namespaced name of snapshot - should be in format "namespace/name"
+      type: string
+  steps:
+    - name: push-snapshot-to-quay
+      image: quay.io/konflux-ci/release-service-utils:066a63d25546239e79533b99c83ff521a045c819
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        push_image() { # Expected arguments are [name, containerImage, repository, tag]
+          # this task is intended to be used for quay.io
+          REGISTRY=quay.io
+
+          SOURCE_AUTH_FILE=$(mktemp)
+          SOURCE_REPO=${2%%@sha256:*}
+          select-oci-auth "$2" | jq -c \
+            '.auths."'"$SOURCE_REPO"'" = .auths."'"$REGISTRY"'" | del(.auths."'"$REGISTRY"'")' > "$SOURCE_AUTH_FILE"
+
+          DEST_AUTH_FILE=$(mktemp)
+          # the auth key will be modified to the full repository path, so that
+          # we can create a combined auth file with source and destination entries for `cosign copy` later
+          select-oci-auth "$3" | jq -c \
+            '.auths."'"$3"'" = .auths."'"$REGISTRY"'" | del(.auths."'"$REGISTRY"'")' > "$DEST_AUTH_FILE"
+
+          DESTINATION_DIGEST=$(oras resolve --registry-config "$DEST_AUTH_FILE" "$3:$4" || true)
+          ORIGIN_DIGEST=$(oras resolve --registry-config "$SOURCE_AUTH_FILE" "$2")
+
+          if [[ "$DESTINATION_DIGEST" != "$ORIGIN_DIGEST" || -z "$DESTINATION_DIGEST" ]]; then
+            # Create a combined auth file to enable partial oci matches to work
+            DOCKER_CONFIG="$(mktemp -d)"
+            export DOCKER_CONFIG
+
+            # SC2016 - "Expanding an array without an index only gives the element in the index 0."
+            # is disabled because "$item" on the next line should only have one element, so there 
+            # is no need to get any index other than 0.
+
+            # shellcheck disable=SC2128
+            jq -s 'reduce .[] as $item ({}; . * $item)' \
+              "$SOURCE_AUTH_FILE" "$DEST_AUTH_FILE" > "$DOCKER_CONFIG"/config.json
+
+            cosign copy -f "$2" "$3:$4"
+
+            echo -n "Pushed image $2 to repository $3:$4"
+          else
+            echo "Component push skipped (source digest exists at destination): $1 $2"
+          fi
+
+        }
+
+        IFS='/' read -r "SNAPSHOT_NAMESPACE" "SNAPSHOT_NAME" <<< "$(params.snapshot)"
+        SNAPSHOT_SPEC=$(kubectl get snapshot "$SNAPSHOT_NAME" -n "$SNAPSHOT_NAMESPACE" \
+          -o jsonpath='{.spec}')
+
+        IFS='/' read -r "RP_NAMESPACE" "RP_NAME" <<< "$(params.releasePlan)"
+        RP_DATA=$(kubectl get releaseplan "$RP_NAME" -n "$RP_NAMESPACE" \
+          -o jsonpath='{.spec.data.mapping}')
```

Comment:
I think at minimum the README should mention that you are getting the location to put the images from the mapping in the ReleasePlan